larq_zoo/literature/meliusnet.py

```diff
@@ -0,0 +1,242 @@
+        x = self.act(x)
+        return tf.keras.layers.Conv2D(
+            filters, 1, use_bias=False, kernel_initializer=self.kernel_initializer
+        )(x)
+
+    def block(self, x):
+        x = self.dense_block(x)
+        return self.improvement_block(x)
+
+    def build(self) -> tf.keras.models.Model:
+        x = self.image_input
+        x = self.group_stem(x)
+        for i, (n, f) in enumerate(zip(self.num_blocks, self.transition_features)):
+            for j in range(n):
+                x = self.block(x)
+            if f:
+                x = self.transition_block(x, f)
+
+        x = self.norm(x)
+        x = self.act(x)
+
+        if self.include_top:
+            x = utils.global_pool(x)
+            x = tf.keras.layers.Dense(
+                self.num_classes, kernel_initializer=self.kernel_initializer
+            )(x)
+            x = tf.keras.layers.Activation("softmax", dtype="float32")(x)
+
+        model = tf.keras.models.Model(
+            inputs=self.image_input, outputs=x, name=self.name
+        )
+
+        if self.weights == "imagenet":
+            if self.include_top:
+                weights_path = self.imagenet_weights_path
+            else:
+                weights_path = self.imagenet_no_top_weights_path
+            model.load_weights(weights_path)
+        elif self.weights is not None:
+            model.load_weights(self.weights)
+
+        return model
+
+
+######################
+# Concrete factories #
+######################
+
+
+@factory
+class MeliusNet22Factory(MeliusNetFactory):
+    num_blocks = (4, 5, 4, 4)
+    transition_features = (160, 224, 256, None)
+    name = "meliusnet22"
+
+    @property
+    def imagenet_weights_path(self):
+        return utils.download_pretrained_model(
+            model="meliusnet22",
+            version="v0.1.0",
+            file="meliusnet22_weights.h5",
+            file_hash="bb8dda20642508bbe5e0ff95012fec450103c4b23989f4c9c9d853d67b6ff806",
+        )
+
+    @property
+    def imagenet_no_top_weights_path(self):
+        return utils.download_pretrained_model(
+            model="meliusnet22",
+            version="v0.1.0",
+            file="meliusnet22_weights_notop.h5",
+            file_hash="9ca867806bff0c2995ff5f1ad085d1627c8dabf12bffbf0bea86eb39ab3cf724",
+        )
+
+
+#########################
+# Functional interfaces #
+#########################
+
+
+def MeliusNet22(
+    *,  # Keyword arguments only
+    input_shape: Optional[Sequence[Optional[int]]] = None,
+    input_tensor: Optional[tf.Tensor] = None,
+    weights: Optional[str] = "imagenet",
+    include_top: bool = True,
+    num_classes: int = 1000,
+) -> tf.keras.models.Model:
+    """Instantiates the MeliusNet22 architecture.
+
+    Optionally loads weights pre-trained on ImageNet.
+
+    ```netron
+    meliusnet22-v0.1.0/meliusnet22.json
+    ```
+    ```plot-altair
+    /plots/meliusnet22.vg.json
+    ```
+    ```summary
+    literature.MeliusNet22
+    ```
+    # ImageNet Metrics
```

Comment:
This needs some spacing due to the table below
```suggestion

    # ImageNet Metrics

```